### PR TITLE
inspec should be a meta role

### PR DESCRIPTION
**Pull request by paulczar**
each role should be responsible for its own inspecs, this enabled that by
setting inspec as a meta role, and demonstrates is use via keystone

### CODE REVIEW

File: roles/inspec/defaults/main.yml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 ---
+install_inspec_controls: []
+
 inspec:
   version: ~
   enabled: true
-  interval: 3600
+  interval: 43200
```
Comment (edtubillara) [inline roles/inspec/defaults/main.yml]: 2x a day sounds good

File: roles/inspec/tasks/main.yml [hunk 2]
```diff
@@ -1,104 +1,136 @@
...
-- name: inspec sensu hook
-  sensu_check: name=inspec-check plugin=check-inspec.rb
-               args='--controls /etc/inspec/host-controls --attrs /etc/inspec/host-controls/attributes.yml' use_sudo=true
-               interval={{ inspec.interval }}
-  notify: restart sensu-client
-  when: inspec.enabled|default(True)|bool
+  - name: fetch inspec dependencies when install method is tar
+    get_url:
+      url: '{{ inspec.profiles[item].url }}'
+      dest: '/etc/inspec/profiles/archive/{{ inspec.profiles[item].name}}-{{ inspec.profiles[item].version }}.tgz'
```
Comment (edtubillara) [inline roles/inspec/tasks/main.yml]: Cool, you've added the .tgz extension.

File: roles/inspec/tasks/main.yml [hunk 3]
```diff
@@ -1,104 +1,136 @@
...
+      mode: 0644
+    with_items: "{{ install_inspec_controls }}"
+
+  - name: inspec attributes
+    template:
+      src: etc/inspec/host-controls/attributes.yml
+      dest: /etc/inspec/attributes.yml
+      mode: 0644
+
+  #- name: inspec openstack control files
```
Comment (edtubillara) [inline roles/inspec/tasks/main.yml]: nit: Is this dead code going to be removed or are we leaving it here for other reasons?

File: roles/inspec/templates/etc/inspec/host-controls/controls/control.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# encoding: utf-8
+# license: Apache 2.0
+# title '{{ item }} controls'
+
+
```
Comment (edtubillara) [inline roles/inspec/templates/etc/inspec/host-controls/controls/control.rb]: I think we are missing a case where we just want to run all the controls and we don't skip any.

1. Required controls is non-empty: use `required_controls` and fill list
2. Skip controls is non-empty: use `include_controls` and fill list with skipped controls
3. Skip controls is empty: do `include_controls` with an empty list

File: roles/inspec/tasks/main.yml (hunk 3 above)
Comment (paulczar) [inline roles/inspec/tasks/main.yml]: removing it ... just wanted to get this working and push something up to get CI'd.

File: roles/inspec/templates/etc/inspec/host-controls/controls/control.rb (hunk 4 above)
Comment (paulczar) [inline roles/inspec/templates/etc/inspec/host-controls/controls/control.rb]: good catch!

File: roles/inspec/templates/etc/inspec/host-controls/controls/control.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+{% elif inspec.controls[item].required_controls|default([])|length > 0 %}
+require_controls "{{ inspec.profiles[inspec.controls[item].profile].name }}" do
+{% for control in inspec.controls[item].required_controls %}
+    control "{{ control }}"
+{% endfor %}
+end
+{% else %}
+include_controls "{{ inspec.profiles[inspec.controls[item].profile].name }}" do
+{% endif %}
+end
```
Comment (edtubillara) [inline roles/inspec/templates/etc/inspec/host-controls/controls/control.rb]: This `end` needs to be moved into the {% endif %}

**Comment by nirajdp76** [comment]
retest

**Comment by nirajdp76** [comment]
retest

**Pull request merged**